proto/spacemesh/global_state.proto

```diff
@@ -3,17 +3,48 @@ package spacemesh;
 import "google/protobuf/empty.proto";
 import "spacemesh/types.proto";
 
-// Readonly global state data.
+// Readonly global state data - current and historical.
 // Global state data is data which is not explicitly stored in the mesh.
 // Global state is modified only by the state transition function.
 service GlobalStateService {
-    //// Streaming global state data
+    // Latest computed global state - layer and its root hash
+    rpc GlobalState(google.protobuf.Empty) returns (GlobalState);
 
-    // Account changes (e.g., balance and counter/nonce changes).
-    rpc AccountStream(google.protobuf.Empty) returns (stream Account);
+    // Get the current transaction state. Note that tx state is not part of
+    // global state. It is metadata about a transaction on its journey to the
+    // mesh and from the mesh to the global state (or to null)
+    rpc TransactionState(TransactionId) returns (TransactionState);
+
+    // Account info in the current global state
+    rpc Account(AccountId) returns (Account);
+
+    // Rewards awarded to an account
+    rpc AccountRewards(AccountIdWithOffset) returns (Rewards);
+
+    // Transaction recipts for transactions to/from an account
+    rpc AccountTransactionReceipts(AccountIdWithOffset) returns (TransactionReceipts);
+
+    // Rewards awarded to a smesher
+    rpc SmesherRewards(SmesherIdWithOffset) returns (Rewards);
+
+    ////////// Streams
+
+    // Rewards awarded to a specific account
+    rpc AccountRewardStream(AccountId) returns (stream Reward);
+
+    // Rewards awarded to a smesher id
+    rpc SmesherRewardStream(SmesherId) returns (stream Reward);
```

Comment:
no - this will be very wasteful. Wallets are only interested in a small subset of rewards, not that 50 or so rewards every 5 minutes that the protocol computes! So, this is needed to get rewards that the wallet knows it needs to display in the future without polling. I'm not sure this can be done otherwise. We need 2 methods because users may change the coinbase account for rewards and want to know of all rewards to their miner regardless of these changes...

Reply:
The requirements for the wallet makes sense, but I still think it may be easier to write this filtering logic on the frontend than inside go-spacemesh. We might want to put this question before the other devs, in a design review.

Reply:
Let's take this to #26

Reply:
ok